services/121-service/src/payments/reconciliation/onafriq-reconciliation/onafriq-reconciliation.service.ts

```diff
@@ -0,0 +1,83 @@
+import { Inject, Injectable } from '@nestjs/common';
+import { Redis } from 'ioredis';
+import { Equal } from 'typeorm';
+import { QueryDeepPartialEntity } from 'typeorm/query-builder/QueryPartialEntity';
+
+import { OnafriqTransactionEntity } from '@121-service/src/payments/fsp-integration/onafriq/entities/onafriq-transaction.entity';
+import { OnafriqTransactionCallbackDto } from '@121-service/src/payments/reconciliation/onafriq-reconciliation/dtos/onafriq-transaction-callback.dto';
+import { OnafriqTransactionCallbackJobDto } from '@121-service/src/payments/reconciliation/onafriq-reconciliation/dtos/onafriq-transaction-callback-job.dto';
+import {
+  getRedisSetName,
+  REDIS_CLIENT,
+} from '@121-service/src/payments/redis/redis-client';
+import { TransactionStatusEnum } from '@121-service/src/payments/transactions/enums/transaction-status.enum';
+import { TransactionEntity } from '@121-service/src/payments/transactions/transaction.entity';
+import { TransactionScopedRepository } from '@121-service/src/payments/transactions/transaction.repository';
+import { QueuesRegistryService } from '@121-service/src/queues-registry/queues-registry.service';
+import { ScopedRepository } from '@121-service/src/scoped.repository';
+import { JobNames } from '@121-service/src/shared/enum/job-names.enum';
+import { getScopedRepositoryProviderName } from '@121-service/src/utils/scope/createScopedRepositoryProvider.helper';
+
+@Injectable()
+export class OnafriqReconciliationService {
+  public constructor(
+    @Inject(getScopedRepositoryProviderName(OnafriqTransactionEntity))
+    private readonly onafriqTransactionScopedRepository: ScopedRepository<OnafriqTransactionEntity>,
+    private readonly transactionScopedRepository: TransactionScopedRepository,
+    private readonly queuesService: QueuesRegistryService,
+    @Inject(REDIS_CLIENT)
+    private readonly redisClient: Redis,
+  ) {}
+
+  public async processTransactionCallback(
+    onafriqTransactionCallback: OnafriqTransactionCallbackDto,
+  ): Promise<void> {
+    const onafriqTransactionCallbackJob: OnafriqTransactionCallbackJobDto = {
+      thirdPartyTransId: onafriqTransactionCallback.thirdPartyTransId,
+      statusCode: onafriqTransactionCallback.status.code,
+      statusMessage: onafriqTransactionCallback.status.message,
+    };
+
+    const job = await this.queuesService.onafriqTransactionCallbackQueue.add(
+      JobNames.default,
+      onafriqTransactionCallbackJob,
+    );
+
+    await this.redisClient.sadd(getRedisSetName(job.data.programId), job.id);
+  }
+
+  public async processOnafriqTransactionCallbackJob(
+    onafriqTransactionCallbackJob: OnafriqTransactionCallbackJobDto,
+  ): Promise<void> {
+    const onafriqTransaction =
```

Comment:
Sorry another suggestion here: 
```suggestion
    const { transactionId } =
```
I did not test this but something like this is cleaner

Reply:
Than you also need to change it at line 79 of course